src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -651,11 +651,13 @@ trait TypeDiagnostics {
     }
     object skipMacroExpansion extends UnusedPrivates {
       override def traverse(t: Tree): Unit =
-        if (!hasMacroExpansionAttachment(t)) super.traverse(t)
+        if (!hasMacroExpansionAttachment(t) && !(t.hasSymbol && isExpanded(t.symbol)))
+          super.traverse(t)
     }
     object checkMacroExpandee extends UnusedPrivates {
       override def traverse(t: Tree): Unit =
-        super.traverse(if (hasMacroExpansionAttachment(t)) macroExpandee(t) else t)
+        if (!(t.hasSymbol && isExpanded(t.symbol)))
+          super.traverse(if (hasMacroExpansionAttachment(t)) macroExpandee(t) else t)
```

Comment:
I think the condition is flipped? This is the "yes, lint the expandee".

Moving the test to neg and including a Test_3 with `-Wmacro:after` to show the error would verify.

Reply:
I only just glanced at the code comment in `StdAttachments` about "should be a better way".

Reply:
I agree having a "pos" test for `-Wmacro:before` and a "neg" test is a good idea, but I'm a bit confused by your suggestion. Is the semantics of neg tests that all the files except the last one *should* compile, and the last one should not compile? So we can write both the pos and the neg parts in the same test?

Reply:
Yes, it would compile A_1, then B_2, then error on C_3, and each round can use different flags. If you think that is too confusing or subtle, pos and neg tests is OK.

Reply:
Thanks for the clarification. I added a separate neg test as I think that's slightly clearer.

Reply:
Not obvious from my first comment, but for "expandee" I always conflate "tree to be expanded" and "result of expansion". I would be less confused by "pre" and "post" or "original" and "expanded", etc.

Reply:
I agree, the naming in this area is mighty confusing.

Reply:
Some followup to my confusion in https://github.com/scala/scala/pull/10693 (in case I have to research the nomenclature again in future). The other naming that confuses me still is each instance of `UnusedPrivates` as it relates to `-Wmacros` settings. Maybe I only have trouble with names I didn't come up with myself. For an improved default behavior for `-Wmacros`, I could only come up with `default`.